src/library/scala/collection/SeqView.scala

```diff
@@ -141,6 +141,8 @@ object SeqView {
     def length: Int = underlying.length
     def iterator: Iterator[A] = Iterator.empty ++ _sorted.iterator // very lazy
     override def knownSize: Int = underlying.knownSize
+    override def isEmpty: Boolean = underlying.isEmpty
+    override def to[C1](factory: Factory[A, C1]): C1 = _sorted.to(factory)
 
     override def sorted[B1 >: A](implicit ord1: Ordering[B1]): SeqView[A] =
       if (ord1 == this.ord) this else super.sorted(ord1)
```

Comment:
could do also

```scala
else if (ord1.isReverseOf(this.ord)) this.reversed else ...
```

Reply:
that's an interesting possibility; it loses laziness though

Reply:
other possibilities include returning a `new SeqView.Reverse(this)`, which of course prevents the optimisation from being done more than once

Reply:
Howso? 

```
view.sorted.reverse
```
shouldn't trigger evaluation

Reply:
> other possibilities include returning a `new SeqView.Reverse(this)`, which of course prevents the optimisation from being done more than once

Could always just new up a custom one:

```scala
val self = this
new SeqView.Reverse(this) {
  override def reverse = self
  override def sorted(implicit ord: Ordering[A]) = 
    if (ord.isReverseOf(ord1)) this 
    else if (ord == ord1) self
    else new SeqView.Sorted(underlying)(ord)
}
```